src/mixins/socket.js

```diff
@@ -129,6 +129,18 @@ export default {
                 this.allowLoginDialog = false;
             });
 
+            socket.on("loginRequired", () => {
+                this.loggedIn = false;
+
+                let token = this.storage().token;
+                if (token && token !== "autoLogin") {
+                    this.loginByToken(token);
+                } else {
+                    this.$root.storage().removeItem("token");
```

Comment:
Previously this was behind a `setTimeout`.
I don't quite get said part in the old code.
Why do you think was this added?

Reply:
Basically this would be exactly the same use case I'm trying to solve here:

The client assumes it's an autoLogin session, but the server requires login.

To get this case in the current WebUI:
1. Set Uptime Kuma to autoLogin
2. Open Uptime Kuma in a browser
3. Now on another device/browser/private session enable login in Uptime Kuma
4. Refresh the page in the original browser

The client now thinks autoLogin is enable but the server expects a username/password -> You get a 5 second "empty" page until this timeout kicks in.

With these changes it wouldn't be required to clear the "wrong" token anymore, but I think it's still nicer not to just leave it in the localStorage.